src/catch2/catch_config.cpp

```diff
@@ -80,6 +81,26 @@ namespace Catch {
             } );
         }
 
+#if defined( CATCH_CONFIG_BAZEL_SUPPORT )
+        // Register a JUnit reporter for Bazel. Bazel sets an enviorment
+        // variable with the path to XML output. If this file is written to
+        // during test, Bazel will not generate a default XML output.
+        // This allows the XML output file to contain higher level of detail
+        // than what is possible otherwise.
+#if defined( CATCH_PLATFORM_WINDOWS )
+        // On Windows getenv throws a warning as there is no input validation,
+        // since the key is hardcoded, this should not be an issue.
+#pragma warning(suppress: 4996)
+        auto const bazelOutputFile = std::getenv( "XML_OUTPUT_FILE" );
+#else
+        auto const bazelOutputFile = std::getenv( "XML_OUTPUT_FILE" );
+#endif
```

Comment:
Why not have just the warning suppression behind the if-def?

Reply:
I am also not a fan of unterminated warning suppression.

Reply:
That was my original approach but annoyingly the `suppress` applies exactly to the next line, which was `#endif`. 
Could you let me know what you mean by 'unterminated warning suppression'? `suppress` pushes the pragma on the stack and pops the warning stack so the state is reset afterwards. It will only apply to line 94. 

I wasn't a big fan of the duplicated line here either but thought maybe someone smarter than me would implement the MSC envrioment getter better than this in the future.

Reply:
`suppress` is actually new to me, I am used to using `disable`.